care/facility/tasks/asset_monitor.py

```diff
@@ -45,42 +49,43 @@ def check_asset_status():
                             "middleware_hostname": hostname,
                         }
                     )
+                    # Fetching the status of the device
                     result = asset_class.api_get(asset_class.get_url("devices/status"))
-                    middleware_status_cache[hostname] = result
                 except Exception:
-                    logger.exception("Error in Asset Status Check - Fetching Status")
-                    middleware_status_cache[hostname] = None
-                    continue
+                    logger.warn(f"Middleware {hostname} is down", exc_info=True)
 
-            if not result:
-                continue
+            # If no status is returned, setting default status as down
+            if not result or len(result) == 0:
```

Comment:
```suggestion
            if not result:
```
`not {}` returns true so no need to do length check

Reply:
yes `not []` also returns true

Reply:
Oh yea, How did I miss that. 🤔